Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -210,29 +210,34 @@ void PeakFitResult::setBadRecord(size_t ipeak, const double peak_position) {
 } // namespace FitPeaksAlgorithm
 
 //----------------------------------------------------------------------------------------------
+namespace {
+
 /** Get an index of a value in a sorted vector.  The index should be the item
  * with value nearest to X
  */
-size_t findXIndex(const std::vector<double> &vecx, double x) {
+template <typename vector_like>
+size_t findXIndex(const vector_like &vecx, const double x,
+                  const size_t startindex = 0) {
   size_t index;
   if (x <= vecx.front()) {
     index = 0;
   } else if (x >= vecx.back()) {
     index = vecx.size() - 1;
   } else {
-    auto fiter = lower_bound(vecx.begin(), vecx.end(), x);
-    if (fiter == vecx.end())
+    const auto fiter = lower_bound(vecx.cbegin() + startindex, vecx.cend(), x);
```

Comment:
This fix reduces the size of "N". Since `lower_bound` does a bisecting search, it would (in principle) be faster to start from the left side and do a linear search to find the right.